frontend/src/modules/scaffold/views/scaffoldAssetItemView.js

```diff
@@ -0,0 +1,211 @@
+  'modules/assetManagement/collections/assetCollection'
+], function(Origin, BackboneForms, Helpers, ScaffoldAssetView, CourseAssetModel, AssetManagementModalView, AssetCollection) {
+
+  var ScaffoldAssetItemView = ScaffoldAssetView.extend({
+
+    events: function() {
+      return _.extend({}, ScaffoldAssetView.prototype.events, {
+        'click .add-item': 'onAddItemClick',
+        'click .ui-draggable': 'onDragItemClick'
+      });
+    },
+
+  	onAddItemClick: function(event) {
+      event.preventDefault();
+      this.itemListView.addItem();
+    },
+
+    onDragItemClick: function(event) {
+      var item = _.find(this.dragItems, function(i) {
+        return i.elm[0] === event.currentTarget;
+      });
+      item.view.openEditor();
+    },
+
+    initialize: function(options) {
+      ScaffoldAssetView.prototype.initialize.apply(this, arguments);
+      this.init(options);
+    },
+
+    init: function(options) {
+      this.dragItems = [];
+      this.precision = options.schema.editorAttrs['data-precision'] || 3;
+      this.topAttr = options.schema.editorAttrs['data-topAttribute'] || '_top';
+      this.leftAttr = options.schema.editorAttrs['data-leftAttribute'] || '_left';
+      this.itemsPath = options.schema.editorAttrs['data-items'] || 'properties._items';
+      this.itemsPath = this.itemsPath.split('.');
+    },
+
+    postRender: function() {
+      if (!this.itemListView) {
+        this.itemListView = this.findItemView();
+        this.listenTo(this.itemListView, {
+          'add': this.onItemListAdd,
+          'remove': this.onItemListRemove,
+          'change': this.onItemListChange
+        });
+      }
+      // double defer:
+      // if external asset, postRender is triggered before el is injected into Dom
+      _.defer(function() {
+        this.removeDragItems();
+        this.itemListView.items && this.itemListView.items.forEach(function(item) {
+          this.addDragItem(item.editor);
+        }, this);
+      }.bind(this));
+    },
+
+    findItemView: function() {
+      var form = Origin.scaffold.getCurrentForm();
+
+      switch (this.schema.editorAttrs['data-type']) {
+        case 'extension':
+          return form.fields._extensions.editor.nestedForm.fields[this.schema.editorAttrs['data-targetAttribute']].editor.nestedForm.fields[this.schema.editorAttrs['data-items']].editor;
+        case 'menu':
+          return form.fields.menuSettings.editor.nestedForm.fields[this.schema.editorAttrs['data-targetAttribute']].editor.nestedForm.fields[this.schema.editorAttrs['data-items']].editor;
+        default:
+          return form.fields.properties.editor.nestedForm.fields[this.schema.editorAttrs['data-items']].editor;
+      }
+    },
+
+    removeDragItems: function() {
+      this.dragItems.forEach(this.removeDragItem, this);
+      this.dragItems = [];
+    },
+
+    addDragItem: function(itemView) {
+      var value = itemView.value;
+      var index = itemView.$el.closest('.list-item').index();
```

Comment:
Is it possible to switch to utilise a `data-index` attribute or similar instead of the DOM order?

Reply:
We would need to patch scaffoldListView for this.